src/lib/mina_generators/parties_generators.ml

```diff
@@ -1173,9 +1130,26 @@ let gen_parties_from ?failure ?(max_other_parties = max_other_parties)
   let account_ids_seen = Account_id.Hash_set.create () in
   let%bind fee_payer =
     gen_fee_payer ?failure ~permissions_auth:Control.Tag.Signature
-      ~account_id:fee_payer_account_id ~ledger ?vk ~account_state_tbl ()
+      ~account_id:fee_payer_acct_id ?vk ~account_state_tbl ()
+  in
+  let zkapp_account_ids =
+    Account_id.Table.filteri account_state_tbl ~f:(fun ~key:_ ~data:(a, role) ->
+        match role with
+        | `Fee_payer | `New_account ->
+            false
+        | `Ordinary_participant ->
+            Option.is_some a.zkapp )
+    |> Account_id.Table.keys
+  in
+  Hash_set.add account_ids_seen fee_payer_acct_id ;
+  let%bind balancing_party =
+    let authorization = Control.Signature Signature.dummy in
+    gen_party_from ?failure ~permissions_auth:Control.Tag.Signature
```

Comment:
I was doing this because I don't know whether random generated party would increment the nonce or not. Since balancing party would be the head of the list of parties, so I pre-generated it in order to prepare other parties for the change  of nonce.